### PR TITLE
Added new module for denmark locale

### PR DESCRIPTION
**Pull request by odoohouse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_dk/data/account_account_groups.xml [hunk 1]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <!--
+     (c) 2018 Odoo House ApS
+     <field name="type">receivable</field>
+     <field name="type">payable</field>
+     <field name="type">liquidity</field>
+     <field name="type">other</field>
+    -->
+    <data noupdate="True">
```
Comment (sswapnesh) [inline addons/l10n_dk/data/account_account_groups.xml]:
```suggestion
    <data noupdate="1">
```
Everywhere you find this on odoo (Excluding only one place)

File: addons/l10n_dk/data/account_chart_template.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<odoo>
+    <data noupdate="True">
```
Comment (sswapnesh) [inline addons/l10n_dk/data/account_chart_template.xml]:
```suggestion
    <data noupdate="1">
```
Everywhere you find this on odoo (Excluding only one place)

**Comment by mart-e** [comment]
Can you have a look at #29062 ?
I believe the work was continued there.

**Comment by odoohouse** [comment]
@mart-e Can you please help me on this. Bot run executed successfully but merging is blocked for me. What can be the reason?

**Comment by jcr-odoo** [comment]
@jco-odoo Hey, can you check that?

**Comment by jco-odoo** [comment]
@odoohouse We don't merge through the github buttons, so no problem there.  This looks quite good.  

@alyngsoe Would be nice if you could test this one and check.  (if you would not have already, see also tax report in enterprise)

I will check further in the following days.

**Comment by alyngsoe** [comment]
@jco-odoo i already tested locally before odoohouse submitted it.
Is it available at runbot also ? ( i'm not that familier with the whole github-->runbot thing ;-) )

**Comment by mart-e** [comment]
@alyngsoe you can go to http://runbot.odoo.com/runbot/repo/1?search=32964 where you can find the test instances generated for this particular pull request.
Each time a new commit is made in the PR, a new build is created. The 4 last are shown with the colour depending of the status (blue = testing, green = all tests succeeded, red = something was wrong).
These are temporary and will be discarded after a few days.

If you check the dropdown with the gear icon in the last build, you can see a "Connect all" and "Connect base" to connect to an instance with all modules installed on only the base ones. To test a localisation, it is probably better to use the base, to configure properly the company and install the needed modules. By default you have 2 users with the login/password : admin/admin and demo/demo.

**Comment by alyngsoe** [comment]
@jco-odoo I cannot really test, as the runbot is on a CE version. The modules are for enterprise, as far is i'm concerned !

**Comment by jco-odoo** [comment]
@alyngsoe I created a pr https://github.com/odoo/enterprise/pull/6167 you could test on runbot (go to enterprise and search 6167)

But the groups here in the PR are used as tags and not as groups.  I hope these screenshots make it clear: groups we could use with the generic balance sheet in order to show the account hierarchy (need to activate the hierarchy option on the financial report for the balance sheet)  We can easily change the object from account_group to account_tag, it does not change anything to the logic.  (also groups can have parent groups, ...)

Also, we see that the passive on the balance sheet is negative. 

Screenshot from 2019-10-16 10-15-30
Screenshot from 2019-10-16 10-10-55

**Comment by odoohouse** [comment]
@jco-odoo P&L an Balance is testet and also in use by many customers in Denmark. Please leave the groups and not use tags as it is better for customers. It ensure that only one group canbe selected and leave tags for other use fx. key numbers.

**Comment by jco-odoo** [comment]
@odoohouse Nice that this is tested, but the groups vs tags still bothers me.  If the groups were used with the generic balance sheet, we could agree (see images), but here a new report is created and for that, every other localization uses tags.  

If tags are confusing because they are multiple on one account, then it seems like something we should think about in generic accounting and not change country by country.

**Comment by alyngsoe** [comment]
@jco-odoo 
1) We have been using account groups now to manually build the P&L and Balance sheet for several Danish companies since version11, and now also with version12. This works perfectly for us, and i see no real advantage of using tags instead. The only argument i see is that other localizations are not using groups, which is not really an argument to me, as we won't be mixing localizations ! Also, in Denmark we really like to be first movers ;-)
@odoohouse, what is your opinion on this ?

2) JCO points out that the ballance sum is negativ: From an accounting point of view this is actually correct, as we wan't to see the values on the ballance report as: debet = positive / credit = negative.

**Comment by jcr-odoo** [comment]
@alyngsoe @odoohouse 

To clear things up. We are going to transform the account groups into tags and it will work the same way. 
The reason why you should not use account groups is because this feature was not designed for that purpose, it was only designed to have additional subtotals and a hierarchy but this is not working very well for now since its heavy to configure and a bit bugged too. 
The account groups feature will probably get revamped since it is not working very well (for its original purpose). So its more of a lucky punch that it works the same as the tags at the moment.
We will come back to you to test it one last time once we changed the groups to tags.

**Comment by jcr-odoo** [comment]
@alyngsoe @odoohouse 
Hello Again, 

here is the branch where we replaced account groups by account tags. Could you give it a try and validate the functional aspects? 
If somthing is not working as expected, you can ping @rgo-odoo , he will handle it then. 
http://runbot.odoo.com/runbot/repo/git-github-com-odoo-dev-enterprise-8?search=12.0-dk-jco

Thanks

**Comment by alyngsoe** [comment]
@rgo-odoo I don't see the Danish COA and TAX codes in this branch ?

**Comment by rgo-odoo** [comment]
@alyngsoe one need to create a new company and load Danish CoA for it. I did it for you in above branch. You can test it now. Thanks.

**Comment by alyngsoe** [comment]
@jcr-odoo As soon as the branch i accepted, can you also migrate it to v13 ? We are starting to roll-out new v13 customers from next week !

**Comment by alyngsoe** [comment]
@jcr-odoo Any status on this PR ?

**Comment by jco-odoo** [comment]
@alyngsoe I see branches are ready for both 12 and 13.  I just need some time for a quick review.

**Comment by alyngsoe** [comment]
@jcr-odoo that is excellent ! We are setting up a new customer next week on v13, so that would be a good opportunity to go live with the new localization.

**Comment by alyngsoe** [comment]
@jco-odoo Did you have time to review the branches ? We are setting up several v13 now, and need the new localization quite urgently now.

**Comment by jco-odoo** [comment]
@alyngsoe I am surprised there is no VAT Return in it.  Because I see in https://skat.dk/skat.aspx?oid=2244393&ik_navn=transport  that you do file them.

**Comment by alyngsoe** [comment]
@jco-odoo i don't know how to setup VAT return with odoo - never used it. I suggest that we start-up the localization without, and then we will test it for the next PR.

**Comment by jco-odoo** [comment]
@alyngsoe  But as the generic vat report just groups by tax, do you do a sum then to get the numbers you need to report to the government?  In other words, is it sufficiently easy to fill in the numbers there with the generic tax report?  

If there is one we can easily make to have it clearer to the user, I would prefer it, because otherwise people risk to have to link the taxes with the report lines themselves afterwards.  

Furthermore, I saw that the P&L (Resultats...) gives negative sales and positive costs, while if you look in the generic one everything is positive and then in the end the difference is made with a negative.  Also in the balance, but it might be logical there for the nullkontrol somehow, but a negative profit that is a profit is weird.   (but I checked in our branch, need to see if there is any difference with yours)

**Comment by alyngsoe** [comment]
@jco-odoo 
1) Tax report:
It is possible, but not always easy to get the tax amounts to report from the generic tax report. The problem is that we need to sum and add/substract some amount to specific reporting amounts, depending on the type of activities ind the company etc.. I don't have the time to  make a matrix for this now, and would prefer, if we can roll-out the localization modules now. I will then work on these with a specific customer case, to create a new dk_tax report.

2) You are right, we should change the Resultatopg. report to show everything as positive.

**Comment by jco-odoo** [comment]
@alyngsoe 
1 -> This can not be solved easily by creating extra taxes and letting the user select the correct ones on the invoice?  

2 @rgo-odoo I did some changes in your 13.0 (saas-12.3) branch (which would need to be backported still afterwards).  Can you check to change the sign for everything related to sale taxes on the Resultatopg.?  And then the profit should be sales - purchase stuff.

**Comment by jco-odoo** [comment]
@alyngsoe I wonder also if the layout of the P&L/Balance sheet reports could not be improved a bit.

**Comment by alyngsoe** [comment]
@jco-odoo You are welcome to make improvements ;-) BUT, we need to move on now, so we can start using the modules on the new customer implementations, due to start on january 1th 2020 - this means: we need the modules to be published urgently now !

**Comment by jco-odoo** [comment]
@alyngsoe These small changes can go fast.  And if necessary, we can quickly merge without VAT report.  There is no problem there.  We want to advance too. 

But on the other hand, it would be good to have a better idea what is necessary for the VAT report.  Because normally, a tax can always be linked to a certain tax grid.  And here it is weird that the tax grid would differ depending on your activity.  It would be nice if you could refer to more documentation about it.

**Comment by jco-odoo** [comment]
@alyngsoe If you want, if you can just confirm the changes for the profit and loss, it can be merged.

**Comment by alyngsoe** [comment]
@jco-odoo where do i test the profit and lost changes ?

**Comment by rgo-odoo** [comment]
@alyngsoe you can connect to latest build here, http://runbot.odoo.com/runbot/repo/git-github-com-odoo-dev-enterprise-8?search=13.0-l10n_dk-rgo , where I have loaded Denmark localization

**Comment by alyngsoe** [comment]
@jco-odoo i have just testet, and can confirm the changes :-) Please merge the branch, so we can go live with it :-)

**Comment by C3POdoo** [comment]
Dear @odoohouse,

Thank you for your contribution but the version 12.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**